packages/synapse-react-client/src/components/AclEditor/useUpdateAcl.ts

```diff
@@ -35,57 +40,64 @@ export default function useUpdateAcl(options: UseUpdateAclOptions) {
-    newReviewerId: string | null,
-    accessTypes: ACCESS_TYPE[],
-  ) => {
-    setIsDirty(true)
-    if (newReviewerId) {
-      const alreadyReviewer = resourceAccessList.some(
-        resourceAccess => resourceAccess.principalId === Number(newReviewerId),
-      )
-      if (alreadyReviewer) {
-        onError(PRINCIPAL_ALREADY_ADDED_ERROR_MESSAGE)
-      } else {
-        const newResourceAccess: ResourceAccess = {
-          principalId: Number(newReviewerId),
-          accessType: accessTypes,
-        }
-        const updatedResourceAccessList = [
-          ...resourceAccessList,
-          newResourceAccess,
-        ]
-        setResourceAccessList(updatedResourceAccessList)
+  const addResourceAccessItem = useCallback(
+    (newReviewerId: string | null, accessTypes: ACCESS_TYPE[]) => {
+      setIsDirty(true)
+      if (newReviewerId) {
+        setResourceAccessList(resourceAccessList => {
+          const alreadyReviewer = resourceAccessList.some(
+            resourceAccess =>
+              resourceAccess.principalId === Number(newReviewerId),
+          )
+          if (alreadyReviewer) {
+            onError(PRINCIPAL_ALREADY_ADDED_ERROR_MESSAGE)
+          } else {
+            const newResourceAccess: ResourceAccess = {
+              principalId: Number(newReviewerId),
+              accessType: accessTypes,
+            }
+            return [...resourceAccessList, newResourceAccess]
+          }
+          return resourceAccessList
+        })
       }
-    }
-  }
+    },
+    [onError],
+  )
 
-  const updateResourceAccessItem = (
-    principalId: number,
-    accessType: ACCESS_TYPE[],
-  ) => {
-    setIsDirty(true)
-    const updatedResourceAccessList = resourceAccessList.map(resourceAccess => {
-      return resourceAccess.principalId === principalId
-        ? { ...resourceAccess, accessType }
-        : resourceAccess
-    })
-    setResourceAccessList(updatedResourceAccessList)
-  }
+  const updateResourceAccessItem = useCallback(
+    (principalId: number, accessType: ACCESS_TYPE[]) => {
+      setIsDirty(true)
+      setResourceAccessList(resourceAccessList =>
+        resourceAccessList.map(resourceAccess => {
+          return resourceAccess.principalId === principalId
+            ? { ...resourceAccess, accessType }
+            : resourceAccess
+        }),
+      )
+    },
+    [],
+  )
 
-  const removeResourceAccessItem = (principalId: number) => {
+  const removeResourceAccessItem = useCallback((principalId: number) => {
     setIsDirty(true)
-    const updatedResourceAccessList = resourceAccessList.filter(
-      raListItem => raListItem.principalId !== principalId,
+    setResourceAccessList(resourceAccessList =>
+      resourceAccessList.filter(
+        raListItem => raListItem.principalId !== principalId,
+      ),
     )
-    setResourceAccessList(updatedResourceAccessList)
-  }
+  }, [])
+
+  const resetDirtyState = useCallback(() => {
+    setIsDirty(false)
+  }, [])
```

Comment:
TODO: Add a return type and comment so it's clear that `resetDirtyState` basically exists to re-sort.